src/algebra/README.md

```diff
@@ -0,0 +1,25 @@
+# Algebra
+
+[Groups](https://en.wikipedia.org/wiki/Group_(mathematics)) $G$ are algebraic structures which are set and has a binary operation $\oplus$ that combines two elements $a, b$ of the set to produce a third element $a\oplus b$ in the set.
+The operation is said to have following properties:
+1. Closure: $a\oplus b=c\in G$
+2. Associative: $(a\oplus b)\oplus c = a\oplus(b\oplus c)$
+3. Existence of Identity element: $a\oplus 0 = 0\oplus a = a$
+4. Existence of inverse element for every element of the set: $a\oplus b=0$
+5. Commutativity: Groups which satisfy an additional property: *commutativity* on the set of
+   elements are known as **Abelian groups**.
```

Comment:
This is a nitpick, so don't feel compelled to change this (i love your documentation always) but I have a disdain for the $\oplus$ as a group op. Why? 

$\oplus$ is used, usually, as a *direct sum* or *coproduct* so it is a bit weird. It also carries connotation of commutivity which is usually *not* guaranteed in groups! (most groups are not commutative)

Instead consider just using concatenation: $ab = c$, $(ab)c=a(bc)$, etc.

Reply:
I need to learn more abstract algebra :)

I think concatenation is mostly substituted as multiplication, at least in my head, so might create confusion. does using `\cdot` make more sense?

Found it used in Ben Lynn's notes as well: https://crypto.stanford.edu/pbc/notes/group/group.html

Reply:
modified it to $\cdot$, does this work?

if not can just simplify it to concatenation

Reply:
$\cdot$ is fine. For a group, we just don't want to convey the operation is + as that implies commutativity in math people's heads. Concatenation is actually really general as you are thinking of group elements as strings of chars with rules in how to reduce the strings into other strings. Strings of minimal length given the reduction rules are considered canonical representatives (this, at least, is the way [group presentations](https://en.wikipedia.org/wiki/Presentation_of_a_group) define a group). All groups can be given via a presentation, albeit that they may not be finite presentations (that is really not a problem for us lol)